### PR TITLE
[ADD] sale_order_line_effective_date

### PR DESCRIPTION
**Pull request by EmilioPascual**
Add field effective date in sale order line. This field is a compute with completion date of the first delivery.

@yajo @Shide @rafaelbn @Gelojr please review

MT-6075 @moduon

### CODE REVIEW

File: sale_order_line_effective_date/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        help="Completion date of the first delivery order.",
+    )
+
+    @api.depends("move_ids.date")
+    def _compute_effective_date(self):
+        for line in self:
+            moves = line.move_ids.filtered(
+                lambda x: x.state == "done" and x.location_dest_id.usage == "customer"
+            )
+            dates_list = [date for date in moves.mapped("date") if date]
```
Comment (yajo) [inline sale_order_line_effective_date/models/sale_order_line.py]: suggestion: filtering twice feels awkward

```suggestion
            moves = line.move_ids.filtered(
                lambda x: x.state == "done" and x.location_dest_id.usage == "customer" and x.date
            )
            dates_list = moves.mapped("date")
```

File: sale_order_line_effective_date/readme/DESCRIPTION.md [hunk 2]
```diff
@@ -0,0 +1 @@
+This module extends the functionality of sale stock and add field effective date in sale order line, this field is completion date of the first delivery order.
```
Comment (yajo) [inline sale_order_line_effective_date/readme/DESCRIPTION.md]:
```suggestion
This module extends the functionality of sale stock and adds the field "effective date" in sale order lines.
This field displays the completion date of the first delivery order.
```

File: sale_order_line_effective_date/readme/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+To use this module, you need to:
+
+1. Go to Sales > Orders > Quotation.
+1. Create new quotation with storable product.
```
Comment (yajo) [inline sale_order_line_effective_date/readme/USAGE.md]:
```suggestion
1. Create a new quotation with a storable product.
```

File: sale_order_line_effective_date/readme/USAGE.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+1. Confirm quotation.
+1. Click on smart button delivery.
+1. Set quantities done in delivery picking.
+1. Validate delivery picking.
+
+To see field effective date you can install *sale_order_line_menu*
```
Comment (yajo) [inline sale_order_line_effective_date/readme/USAGE.md]: This would fit better in `INSTALL.md`

**Review by yajo — CHANGES_REQUESTED** [review]

File: sale_order_line_effective_date/models/sale_order_line.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            line.effective_date = min(dates_list, default=False)
```
Comment (Shide) [inline sale_order_line_effective_date/models/sale_order_line.py]: Use [:1] to get an empty Recordset if there is no records in the Recordset and .date will give you False

```suggestion
            line.effective_date = line.move_ids.filtered_domain([
                ("state", "=", "done"),
                ("location_dest_id.usage", "=", "customer"),
                ("date", "!=", False),
            ]).sorted("date")[:1].date
```

**Review by Shide — CHANGES_REQUESTED** [review]
*nitpick*: Simplification

**Comment by Shide** [comment]
Added another field: Last Effective Date (last delivery date)

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
@gelojr please review

**Review by Shide — APPROVED** [review]

**Review by Gelojr — APPROVED** [review]
LGTM Thanks @Shide

**Pull request merged**